server/game/cards/02_SHD/units/XanaduBloodCadBanesReward.ts

```diff
@@ -0,0 +1,38 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { RelativePlayer, Trait, ZoneName } from '../../../core/Constants';
+
+export default class XanaduBloodCadBanesReward extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '5818136044',
+            internalName: 'xanadu-blood#cad-banes-reward',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addTriggeredAbility({
+            title: 'Return another friendly non-leader Underworld unit to its owner’s hand',
+            when: {
+                onCardPlayed: (event, context) => event.card === context.source,
+                onAttackDeclared: (event, context) => event.attack.attacker === context.source,
+            },
+            optional: true,
+            targetResolver: {
+                controller: RelativePlayer.Self,
+                cardCondition: (card, context) => card !== context.source && card.isUnit() && card.hasSomeTrait(Trait.Underworld),
+                immediateEffect: AbilityHelper.immediateEffects.returnToHand()
+            },
+            ifYouDo: {
+                title: 'Exhaust an enemy unit or resource',
+                targetResolver: {
+                    controller: RelativePlayer.Opponent,
+                    zoneFilter: [ZoneName.Resource, ZoneName.GroundArena, ZoneName.SpaceArena],
```

Comment:
We don't want to force the player to pick a specific resource to exhaust if they want to choose that option, so we should set this up like the Leia unit where they have a selection with two options.

It actually would be really good to have a filter that will prevent the system from being able to target stuff in the resource zone. Do you mind going to `ExhaustOrReadySystem.canAffect` and removing line 17 (resource zone check), I think that will do it

Reply:
removing this line break some system so I revert change on it
see : https://github.com/SWU-Karabast/forceteki/actions/runs/11900082668